admin/services/data-access/pupils-not-taking-check.data.service.js

```diff
@@ -45,7 +45,9 @@ const pupilsNotTakingCheckDataService = {
       FROM ${sqlService.adminSchema}.[pupil] p 
       LEFT JOIN ${sqlService.adminSchema}.[pupilAttendance] pa ON p.id = pa.pupil_id AND pa.isDeleted=0
       LEFT JOIN ${sqlService.adminSchema}.[pupilGroup] pg ON p.id = pg.pupil_id
-      WHERE p.school_id = @schoolId AND pa.id IS NULL
+      WHERE p.school_id = @schoolId
+      AND p.pupilStatus_id = 1
```

Comment:
We shouldn't use IDs as we don't know what order they be inserted on a particular env.  Please replace with a subquery to fetch the id.

IDs could be safely memo'ised by the app if needed.

Reply:
@jon-shipley I raised a similar concern in a past PR and the change was not made for performance reasons.
I made the change here so the query is updated.